scripts/OverlaySurvey.py

```diff
@@ -0,0 +1,194 @@
+# importing the requests library 
+import requests 
+import json
+import time
+import sys
+import os
+import argparse
+import networkx as nx
+from networkx import Graph
+from collections import defaultdict
+
+def add_new_node(label):
+    if G.has_node(label):
+        return
+    G.add_node(label, label=label)
+
+def add_new_edge(u, v):
+    if G.has_edge(u, v):
+        return
+    G.add_edge(u, v)
+
+def get_next_peers(topology):
+    results = []
+    for key in topology:
+        if topology[key] is None:
+                continue
+
+        if "inboundPeers" in topology[key] and topology[key]["inboundPeers"]:
+            for in_peers in topology[key]["inboundPeers"]:
+                results.append(in_peers["nodeId"])
+        if "outboundPeers" in topology[key] and topology[key]["outboundPeers"]:
+            for out_peers in topology[key]["outboundPeers"]:
+                results.append(out_peers["nodeId"])
+    return results
+
+def send_requests(peer_list):
+    for key in peer_list:
+        PARAMS["node"] = key
+        requests.get(url = SURVEY_REQUEST, params = PARAMS) 
+
+def check_results():
+    r = requests.get(url = SURVEY_RESULT)
+    data = r.json()
+
+    if("topology" not in data):
+        return []
+
+    topology = data["topology"]
+
+    for key in topology:
+        if topology[key] is None:
+            continue
+
+        if "inboundPeers" in topology[key] and topology[key]["inboundPeers"]:
+            for in_peers in topology[key]["inboundPeers"]:
+                merged_results[key]["inboundPeers"][in_peers["nodeId"]] = in_peers
+                add_new_node(key)
+                add_new_node(in_peers["nodeId"])
+                add_new_edge(in_peers["nodeId"], key)
+            merged_results[key]["totalInbound"] = topology[key]["numTotalInboundPeers"]
+        if "outboundPeers" in topology[key] and topology[key]["outboundPeers"]:
+            for out_peers in topology[key]["outboundPeers"]:
+                merged_results[key]["outboundPeers"][out_peers["nodeId"]] = out_peers
+                add_new_node(key)
+                add_new_node(out_peers["nodeId"])
+                add_new_edge(key, out_peers["nodeId"])
+            merged_results[key]["totalOutbound"] = topology[key]["numTotalOutboundPeers"]
+    
+        merged_results[key]["responseSeen"] = True
+
+    return get_next_peers(topology)
+
+def write_graph_stats():
+    stats = {}
+    stats["average_shortest_path_length"] = nx.average_shortest_path_length(G)
+    stats["average_clustering"] = nx.average_clustering(G)
+    stats["clustering"] = nx.clustering(G)
+    stats["degree"] = dict(nx.degree(G))
+    with open('graphStats.json', 'w') as outfile:
```

Comment:
make input and output files command-line args

Reply:
done